src/arithmetic/func_desc.c

```diff
@@ -45,17 +33,17 @@ AR_FuncDesc *AR_FuncDescNew(const char *name, AR_Func func, uint min_argc, uint
 /* Register an arithmetic function. */
 void AR_RegFunc(AR_FuncDesc *func) {
 	char lower_func_name[32] = {0};
-	short lower_func_name_len = 32;
-	_toLower(func->name, &lower_func_name[0], &lower_func_name_len);
+	size_t lower_func_name_len = 32;
+	str_tolower(func->name, lower_func_name, &lower_func_name_len);
 	assert(raxInsert(__aeRegisteredFuncs, (unsigned char *)lower_func_name, lower_func_name_len, func,
 					 NULL) == 1);
 }
 
 /* Get arithmetic function. */
 AR_FuncDesc *AR_GetFunc(const char *func_name) {
 	char lower_func_name[32] = {0};
```

Comment:
This is a good opportunity to remove this initialisation
`char lower_func_name[32];`
Please apply throughout the modified files.